modules/common/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/common/SqsMessage.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2020 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and
+ * limitations there under.
+ */
+package com.snowplowanalytics.snowplow.storage.spark
+
+import io.circe._
+import io.circe.syntax._
+import io.circe.parser
+import io.circe.generic.semiauto._
+
+import cats.implicits._
+
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaVer, SelfDescribingData}
+import com.snowplowanalytics.iglu.core.circe.implicits._
+
+/** A SqsMessage is sent by shredder and read by loader. It contains all the shredded types with their path and type (TSV, JSON) */
+final case class SqsMessage(
+  shreddedTypes: List[ShreddedType]
+) {
+  lazy val schemaKey =
+    SchemaKey(
+      "com.snowplowanalytics.snowplow.storage.rdbshredder",
+      "output",
+      "jsonschema",
+      SchemaVer.Full(1, 0, 0)
+    )
+
+  // Encoders are used by shredder
+  implicit val shreddedFormatEncoder = deriveEncoder[ShreddedFormat]
```

Comment:
Why encoders are inside case class? I'm even surprised it works.

Reply:
It's because we need `.asJson` [here](https://github.com/snowplow/snowplow-rdb-loader/blob/feature/shredder_sqs/modules/common/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/common/SqsMessage.scala#L44)

Reply:
I'm pretty sure it doesn't require encoders inside case class.

```scala
import Foo._

case class Foo(a: Int) {
  lazy val compact = (this: Foo).asJson
}

object Foo {
  implicit val codec: Encoder[Foo] = deriveEncoder[Foo]
}
```